app/controllers/api/v1/user_trails_controller.rb

```diff
@@ -0,0 +1,17 @@
+class Api::V1::UserTrailsController < ApplicationController
+  def create
+    if !user_valid
+      render json: { 'error': 'No User Found' }, status: 200
+    else 
+      UserTrail.create(user_id: params[:user_id], trail_id: params[:trail_id])
```

Comment:
ActiveRecord is awesome in that it will capture a lot of common issues for us. Splitting up the create into .new and .save will allow you to catch ActiveRecord errors more easily. AR would also error out if the user ID didn't exist so if you do it in this way, you can remove your `user_valid` method. 

```
     user_trail = UserTrail.new(user_id: params[:user_id], trail_id: params[:trail_id])
     if user_trail.save
         --> success case
     else
        --> error case // you can use user_trail.errors.full_messages to capture the error messages from ActiveRecord